care/facility/api/viewsets/patient.py

```diff
@@ -26,3 +32,59 @@ def get_serializer_class(self):
             return PatientDetailSerializer
         else:
             return self.serializer_class
+
+    @action(detail=True, methods=["get"])
+    def history(self, request, *args, **kwargs):
+        user = request.user
+        queryset = PatientAdmission.objects.filter(patient__id=self.kwargs.get("pk"))
+        if not user.is_superuser:
+            queryset = queryset.filter(patient__created_by=user)
+        return Response(data=PatientAdmissionSerializer(queryset, many=True).data)
+
+
+class PatientAdmissionViewSet(viewsets.ModelViewSet):
+    permission_classes = (IsAuthenticated,)
+    queryset = PatientAdmission.objects.all()
+    serializer_class = PatientAdmissionSerializer
+    http_method_names = ["put", "delete", "get"]
+
+    def get_queryset(self):
+        user = self.request.user
+        queryset = self.queryset.filter(facility__id=self.kwargs.get("facility_pk"), is_active=True)
+        if user.is_superuser:
+            return queryset
+
+        return queryset.filter(facility__created_by=user)
+
+    def update(self, request, *args, **kwargs):
+        data = {"facility_id": self.kwargs.get("facility_pk"), "patient_id": self.kwargs.get("pk")}
+        admission_date = request.data.get("admission_date")
+        if admission_date:
```

Comment:
In some case the entry may be made while discharging. We cannot expect everything to be done in a seamless way. So we should be allowing to enter the previous data, and hence `discharge_date`